zetaclient/zetabridge/broadcast_test.go

```diff
@@ -1,25 +1,86 @@
 package zetabridge
 
 import (
-	"fmt"
-	"regexp"
+	"encoding/hex"
+	"errors"
+	"net"
+	"testing"
 
-	. "gopkg.in/check.v1"
+	"github.com/cosmos/cosmos-sdk/types"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
+	feemarkettypes "github.com/evmos/ethermint/x/feemarket/types"
+	"github.com/stretchr/testify/require"
+	"github.com/zeta-chain/zetacore/pkg/chains"
+	crosschaintypes "github.com/zeta-chain/zetacore/x/crosschain/types"
+	observerTypes "github.com/zeta-chain/zetacore/x/observer/types"
+	"github.com/zeta-chain/zetacore/zetaclient/keys"
+	"github.com/zeta-chain/zetacore/zetaclient/testutils/stub"
+	"go.nhat.io/grpcmock"
+	"go.nhat.io/grpcmock/planner"
 )
 
-type BcastSuite struct {
+func TestHandleBroadcastError(t *testing.T) {
+	type response struct {
+		retry  bool
+		report bool
+	}
+	testCases := map[error]response{
+		errors.New("nonce too low"):                       {retry: false, report: false},
+		errors.New("replacement transaction underpriced"): {retry: false, report: false},
+		errors.New("already known"):                       {retry: false, report: true},
+		errors.New(""):                                    {retry: true, report: false},
+	}
+	for input, output := range testCases {
+		retry, report := HandleBroadcastError(input, "", "", "")
+		require.Equal(t, output.report, report)
+		require.Equal(t, output.retry, retry)
+	}
 }
 
-var _ = Suite(&BcastSuite{})
+func TestBroadcast(t *testing.T) {
+	address := types.AccAddress(stub.TestKeyringPair.PubKey().Address().Bytes())
 
-func (s *BcastSuite) SetUpTest(c *C) {
-	fmt.Println("hello")
-}
+	//Setup server for multiple grpc calls
+	l, err := net.Listen("tcp", "127.0.0.1:9090")
```

Comment:
What does `l` means? We should use explicit variable names

Reply:
This is the listener, I will change the name.